src/managers/GitRefManager.js

```diff
@@ -103,6 +103,19 @@ export class GitRefManager {
     const normalizeValue = value => value.trim() + '\n'
     await fs.write(join(gitdir, ref), normalizeValue(value), 'utf8')
   }
+  static async writeSymbolicRef ({ fs: _fs, gitdir, ref, value }) {
+    const fs = new FileSystem(_fs)
+    // TODO: Validate input
```

Comment:
To validate input I'd use 
```js
import cleanGitRef from 'clean-git-ref'

if (ref !== cleanGitRef.clean(ref)) {
  throw new GitError(E.InvalidRefNameError, {
    verb: 'write',
    noun: 'symbolic ref',
    ref,
    suggestion: cleanGitRef.clean(ref)
  })
}

if (value !== cleanGitRef.clean(value)) {
  throw new GitError(E.InvalidRefNameError, {
    verb: 'write',
    noun: 'symbolic ref',
    ref: value,
    suggestion: cleanGitRef.clean(value)
  })
}
```

Reply:
I would put that validation in the user-facing `writeRef` command though. Inside GitRefManager assume input is already validated.